akka-http-core/src/main/scala/akka/http/model/ws/UpgradeToWebsocket.scala

```diff
@@ -32,4 +37,15 @@ trait UpgradeToWebsocket extends HttpHeader {
    * a subprotocol out of the ones requested by the client can be chosen.
    */
   def handleMessages(handlerFlow: Flow[Message, Message, Any], subprotocol: Option[String] = None)(implicit mat: FlowMaterializer): HttpResponse
+
+  import scala.collection.JavaConverters._
+  def getRequestedProtocols(): Iterable[String] = requestedProtocols.asJava
+  def handleMessagesWith(handlerFlow: stream.javadsl.Flow[javadsl.model.ws.Message, javadsl.model.ws.Message, _], materializer: FlowMaterializer): HttpResponse =
+    handleMessages(adaptJavaFlow(handlerFlow))(materializer)
+
+  def handleMessagesWith(handlerFlow: stream.javadsl.Flow[javadsl.model.ws.Message, javadsl.model.ws.Message, _], subprotocol: String, materializer: FlowMaterializer): HttpResponse =
+    handleMessages(adaptJavaFlow(handlerFlow), subprotocol = Some(subprotocol))(materializer)
+
+  private[this] def adaptJavaFlow(handlerFlow: stream.javadsl.Flow[javadsl.model.ws.Message, javadsl.model.ws.Message, _]): Flow[Message, Message, Any] =
+    Flow[Message].map(javadsl.model.ws.Message.adapt).via(handlerFlow.asScala).map(_.asScala)
```

Comment:
the asScala is no longer needed because via accepts any Graph[FlowShape] so you can pass a scaladsl.Flow into the Java APIs and vice versa. You don't need to fix it now, just a heads up.